cartographer/mapping_2d/sparse_pose_graph.cc

```diff
@@ -622,6 +622,7 @@ int SparsePoseGraph::TrimmingHandle::num_submaps(
 
 void SparsePoseGraph::TrimmingHandle::MarkSubmapAsTrimmed(
     const mapping::SubmapId& submap_id) {
+  common::MutexLocker locker(&parent_->mutex_);
```

Comment:
Same here.

Reply:
Because it reads parent_->submap_data_ so I add a locker to lock the parent_mutex_.

Reply:
Yes, but this gets called via line 351 and line 347 already locks the mutex. This looks like a deadlock with a non-recursive mutex to me. Revert?

Reply:
I see, thanks!
I will add REQUIRES(parent_->mutex_) at .h instead.